R/utils.R

```diff
@@ -8,14 +8,39 @@
 #' @usage lhs \%>\% rhs
 NULL
 
-at_selection <- function(nm, .at){
-  if (is_quosures(.at)) {
-    lifecycle::deprecate_warn("0.4.0", I("using `vars()` in .at"))
+at_selection <- function(x, at, error_arg = caller_arg(at), error_call = caller_env()) {
+  if (is_formula(at)) {
+    at <- rlang::as_function(at, arg = error_arg, call = error_call)
+  }
+  if (is.function(at)) {
+    at <- at(names(x))
+  }
 
+  if (is.numeric(at) || is.logical(at) ||is.character(at)) {
+    if (is.character(at) && !is.null(names(x))) {
+      at <- intersect(at, names(x))
+    }
+
+    at <- vec_as_subscript(at, arg = "at", call = error_call)
+    vec_as_location(
+      at,
+      length(x),
+      names(x),
+      missing = "error",
+      arg = "at",
+      call = error_call
+    )
+  } else if (is_quosures(at)) {
+    lifecycle::deprecate_warn("0.4.0", I("using `vars()` in .at"))
     check_installed("tidyselect", "for using tidyselect in `map_at()`.")
-    .at <- tidyselect::vars_select(.vars = nm, !!!.at)
+
+    tidyselect::vars_select(.vars = names(x), !!!at)
+  } else {
+    cli::cli_abort(
+      "{.arg {error_arg}} must be a numeric vector, character vector, or function, not {.obj_type_friendly {at}}.",
+      caller = error_call
```

Comment:
```suggestion
      call = error_call
```

And regenerate snapshots

Noticed this because a snapshot for this references `at_selection()`